.github/workflows/CI.yaml

```diff
@@ -10,65 +10,90 @@ env:
   TEAMCITY_DEV_URL: ${{ secrets.TEAMCITY_DEV_URL }}
   TEAMCITY_USER: ${{ secrets.TEAMCITY_USER }}
   TEAMCITY_PASSWORD: ${{ secrets.TEAMCITY_PASSWORD }}
-  SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
 
 jobs:
-  snyk-test:
+  compile:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: snyk/actions/setup@master
-      - name: Set up JDK 17
-        uses: actions/setup-java@v2
-        with:
-          java-version: '17'
-          distribution: 'adopt'
-      - name: Snyk test dependencies
-        run: snyk test --all-projects --severity-threshold=medium --fail-on=all
-
-  snyk-monitor:
-    runs-on: ubuntu-latest
-    if: github.event_name != 'pull_request'
-    steps:
-      - uses: actions/checkout@v2
-      - uses: snyk/actions/setup@master
-      - name: Set up JDK 17
-        uses: actions/setup-java@v2
-        with:
-          java-version: '17'
-          distribution: 'adopt'
-      - name: Snyk monitor dependencies
-        run: snyk monitor --all-projects --target-reference=${GITHUB_REF}
+      - uses: actions/checkout@v3
+      - uses: ./.github/actions/setup-jdk
+      - uses: ./.github/actions/setup-gradle-cache
+      - name: Compile
+        run: ./gradlew compileJava compileTestJava
 
-  build:
+  code-ql:
     runs-on: ubuntu-latest
+    needs: compile
 
     # required by CodeQL
     permissions:
       security-events: write
       actions: read
 
-    env:
-      DOCKER_ENTERPRISE_DEV_URL: ${{ secrets.DOCKER_ENTERPRISE_DEV_URL }}
-      DOCKER_COMMUNITY_DEV_URL: ${{ secrets.DOCKER_COMMUNITY_DEV_URL }}
-      ENTERPRISE_TAR: enterprise-docker.tar
-      COMMUNITY_TAR: community-docker.tar
-
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
+      - uses: actions/checkout@v3
```

Comment:
Bumped all the versions, got rid of the warnings we had executing the jobs before:

<img width="1535" src="https://user-images.githubusercontent.com/5649971/218121458-649ee725-5d2d-42ac-99a9-409ddf0dc20d.png">